File: Data Structures & Algorithms/maximum-sum-circular-subarray/submission-1.java
class Solution {
    public int maxSubarraySumCircular(int[] nums) {
        int globMax = nums[0], globMin = nums[0];
        int curMax = 0, curMin = 0;
        int total = 0;
        for (int num : nums) {
            curMax = Math.max(curMax + num, num);
            curMin = Math.min(curMin + num, num);

            
            globMax = Math.max(globMax, curMax);
            globMin = Math.min(globMin, curMin);

            total += num;
        }

        return globMax > 0 ? Math.max(globMax, total - globMin) : globMax;
    }
}

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array-ii/submission-1.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        L, R = 0, 0


        while R < len(nums):
            count = 1

            while R + 1 < len(nums) and nums[R + 1] == nums[R]:
                R += 1
                count += 1
            
            for i in range(min(2, count)):
                nums[L] = nums[R]
                L += 1
            
            R += 1
        
        return L
